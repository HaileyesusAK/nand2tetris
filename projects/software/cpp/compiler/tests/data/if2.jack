if (~true) {
	let a = 10;
	while (a > 0) {
		do cmp(a, b);
		let a = a - 1;
	}
}
else {
	do cmp(a, b);
	return a + b;
}
