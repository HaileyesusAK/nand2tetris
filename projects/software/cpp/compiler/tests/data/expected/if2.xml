<ifStatement>
    <keyword> if </keyword>
    <symbol> ( </symbol>
    <expression>
        <term>
            <symbol> ~ </symbol>
            <term>
                <keyword> true </keyword>
            </term>
        </term>
    </expression>
    <symbol> ) </symbol>
    <symbol> { </symbol>
    <statements>
        <letStatment>
            <keyword> let </keyword>
            <identifier> a </identifier>
            <symbol> = </symbol>
            <expression>
                <term>
                    <integerConstant> 10 </integerConstant>
                </term>
            </expression>
            <symbol> ; </symbol>
        </letStatement>
        <whileStatement>
            <keyword> while </keyword>
            <symbol> ( </symbol>
            <expression>
                <term>
                    <identifier> a </identifier>
                </term>
                <symbol> &gt; </symbol>
                <term>
                    <integerConstant> 0 </integerConstant>
                </term>
            </expression>
            <symbol> ) </symbol>
            <symbol> { </symbol>
            <statements>
                <doStatment>
                    <keyword> do </keyword>
                    <identifier> cmp </identifier>
                    <symbol> ( </symbol>
                    <expressionList>
                        <expression>
                            <term>
                                <identifier> a </identifier>
                            </term>
                        </expression>
                        <symbol> , </symbol>
                        <expression>
                            <term>
                                <identifier> b </identifier>
                            </term>
                        </expression>
                    </expressionList>
                    <symbol> ) </symbol>
                    <symbol> ; </symbol>
                </doStatment>
                <letStatment>
                    <keyword> let </keyword>
                    <identifier> a </identifier>
                    <symbol> = </symbol>
                    <expression>
                        <term>
                            <identifier> a </identifier>
                        </term>
                        <symbol> - </symbol>
                        <term>
                            <integerConstant> 1 </integerConstant>
                        </term>
                    </expression>
                    <symbol> ; </symbol>
                </letStatement>
            </statements>
            <symbol> } </symbol>
        </whileStatement>
    </statements>
    <symbol> } </symbol>
    <keyword> else </keyword>
    <symbol> { </symbol>
    <statements>
        <doStatment>
            <keyword> do </keyword>
            <identifier> cmp </identifier>
            <symbol> ( </symbol>
            <expressionList>
                <expression>
                    <term>
                        <identifier> a </identifier>
                    </term>
                </expression>
                <symbol> , </symbol>
                <expression>
                    <term>
                        <identifier> b </identifier>
                    </term>
                </expression>
            </expressionList>
            <symbol> ) </symbol>
            <symbol> ; </symbol>
        </doStatment>
        <returnStatement>
            <keyword> return </keyword>
            <expression>
                <term>
                    <identifier> a </identifier>
                </term>
                <symbol> + </symbol>
                <term>
                    <identifier> b </identifier>
                </term>
            </expression>
            <symbol> ; </symbol>
        </returnStatement>
    </statements>
    <symbol> } </symbol>
</ifStatement>
